### ISSUE TITLE
[3.3.5] Stormwind NPC's William and Donna walking not running

### ISSUE THREAD

**Issue by mario-64** [TrinityCore/TrinityCore#14775]
Donna [http://www.wowhead.com/npc=2532/donna] & William [http://www.wowhead.com/npc=2533/william] from Stormwind simply walk. In 3.3.5, Donna should be running after William because he took her doll. Additionally, both spout various emotes while she chases him. This was discussed in https://github.com/TrinityCore/TrinityCore/issues/10539 with no real resolution. Ultimately we were able to get them both running but the speed was off because eventually she was able to catch up with him, which should not happen.

Rev 3761d2411968
##

**Comment by Aokromes** [comment]
1st law, update your core and db before reporting bugs.

**Comment by mario-64** [comment]
Issue remains

**Comment by tkrokli** [comment]
@mario-64 : notice that you need to post the core hash and DB version from your latest test also.

**Comment by mario-64** [comment]
@tkrokli I'm using the most recent version of your repack

**Comment by tkrokli** [comment]
Oh. I think we should remove these posts then. I should post a new core + DB update (1.5 = May).

**Comment by Nyeriah** [comment]
Repacks are not supported.

**Comment by Kittnz** [comment]
More info about this issue:
https://github.com/TrinityCore/TrinityCore/issues/10539

**Comment by Killyana** [comment]
I'm just wondering, is it hard to remove the WP from 2532 and make a creature_formation and change the movement type from walk to run for 2533?
Emotes and texts can be done by SAI, ooc.
From what I can see the branch 6.x will ends like cata, an empty core used by shitty pv servers if no one wants contribute.

**Comment by mario-64** [comment]
@Nyeriah tkrokli's "repack" is a simple TC compile. Nothing more. It does not invalidate this issue.

**Comment by Killyana** [comment]
WIP
``` sql

DELETE FROM `creature_formations` WHERE `leaderGUID`=79720;
INSERT INTO `creature_formations` (`leaderGUID`,`memberGUID`,`dist`,`angle`,`groupAI`) VALUES
(79720,79720,0,0,2),
(79720,79721,4,0,2);

UPDATE `creature_template` SET `MovementType` = 0 WHERE `entry` = 2532;
UPDATE `creature_addon` SET `path_id` = 0 WHERE `guid` = 79721;

**Comment by mario-64** [comment]
Thanks Killyana. I applied that change but both NPC's still are walking. On the positive side though she doesn't appear to ever catch up to William

**Comment by k59** [comment]
It says **WIP**, work in progress.

**Comment by Killyana** [comment]
@mario-64 to avoid that the follower catch the leader set the dist also in the first line for the leader like here: 4 yards
(79720,79720,4,0,2),
(79720,79721,4,0,2);

**Comment by mario-64** [comment]
@Killyana excellent work! Thank you so much. Looks great to me

**Comment by Killyana** [comment]
For 6.x  spawns are missing for this 2 npcs, anyway here a start:
``` sql
DELETE FROM `creature_formations` WHERE `leaderGUID`=79721;
INSERT INTO `creature_formations` (`leaderGUID`,`memberGUID`,`dist`,`angle`,`groupAI`) VALUES
(79721,79721,4,0,2),
(79721,79720,4,0,2);

DELETE FROM `waypoint_data` WHERE `id`=797210;
INSERT INTO `waypoint_data` ....


UPDATE `creature_template` SET `AIName`='SmartAI' WHERE `entry`=2532;
DELETE FROM `smart_scripts` WHERE `entryorguid`=2532 AND `source_type`=0;
DELETE FROM `smart_scripts` WHERE `entryorguid`=253200 AND `source_type`=9;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(2532,0,0,0,1,0,100,0,10000,10000,30000,30000,80,253200,0,0,0,0,0,1,0,0,0,0,0,0,0,'Donna - ooc - action list'),
(253200,9,0,0,0,0,100,0,0,0,0,0,1,0,0,0,0,0,0,1,0,0,0,0,0,0,0,'Donna - action list - talk1'),
(253200,9,1,0,0,0,100,0,4000,4000,0,0,1,0,0,0,0,0,0,19,2532,50,0,0,0,0,0,'Donna - action list - talk1 for William');

DELETE FROM `creature_text` WHERE `entry` IN (2533, 2532);
INSERT INTO `creature_text` (`entry`,`groupid`,`id`,`text`,`type`,`language`,`probability`,`emote`,`duration`,`sound`,`comment`, `BroadcastTextId`) VALUES
(2533,0,0,'No fair!  Gimme back my gorilla!!',12,7,100,0,0,0,'William', 693),
(2533,0,1,'Gimmie back Jum-jump!',12,7,100,0,0,0,'William', 705),
(2533,0,2,'You better not break Jum-jump!  It''s not fair!',12,7,100,0,0,0,'William', 706),
(2533,0,3,'WAAAHHH!!!',12,7,100,0,0,0,'William', 707),
(2533,0,4,'You''re being mean!  I''m telling Mommy!',12,7,100,0,0,0,'William', 708),
(2533,0,5,'Stop twisting his legs, they aren''t supposed to bend that way!',12,7,100,0,0,0,'William', 709),
(2533,0,6,'You''re breaking him!',12,7,100,0,0,0,'William', 710),
(2533,0,7,'Stop it! You''ll break his legs off!',12,7,100,0,0,0,'William', 711),
(2532,0,0,'Ha ha! I have your Grindgear Gorilla!',12,7,100,0,0,0,'Donna', 50287),
(2532,0,1,'I wonder if your Grindgear Gorilla can jump across the Park!',12,7,100,0,0,0,'Donna', 50288),
(2532,0,2,'I have your gorilla! Nyah nyah!',12,7,100,0,0,0,'Donna', 50289),
(2532,0,3,'I wonder how far the gears turn?',12,7,100,0,0,0,'Donna', 50290),
(2532,0,4,'If you want him back you''re gonna have to beg... like a monkey!!',12,7,100,0,0,0,'Donna', 50291),
(2532,0,5,'Now who''s the crybaby, crybaby!?',12,7,100,0,0,0,'Donna', 50292),
(2532,0,6,'Baby wants his dolly!',12,7,100,0,0,0,'Donna', 50293),
(2532,0,7,'Whoops!',12,7,100,0,0,0,'Donna', 50294);

**Comment by Nyeriah** [comment]
@mario-64  the exact point of not supporting repacks is the fact that the ones distributing them doesn't update them as often as one would be if they were following our repository directly, which leads to bogus issues of things that were already resolved.

After looking into it I noticed this issue in special was still in place either way and that's why I removed my comment.

**Comment by mario-64** [comment]
@Nyeriah I understand. I never use anything but my own compiles or tkrokli's since his are updated frequently and are straight TC anyway.

**Comment by ccrs** [comment]
@Killyana 
````note
[sql.sql] There is no waypoint which links to the waypoint script 229
[sql.sql] There is no waypoint which links to the waypoint script 230
[sql.sql] There is no waypoint which links to the waypoint script 231
[sql.sql] There is no waypoint which links to the waypoint script 232
[sql.sql] There is no waypoint which links to the waypoint script 233
[sql.sql] There is no waypoint which links to the waypoint script 234
[sql.sql] There is no waypoint which links to the waypoint script 235
[sql.sql] There is no waypoint which links to the waypoint script 236
[sql.sql] There is no waypoint which links to the waypoint script 237
[sql.sql] There is no waypoint which links to the waypoint script 238
[sql.sql] There is no waypoint which links to the waypoint script 239
[sql.sql] There is no waypoint which links to the waypoint script 240
[sql.sql] There is no waypoint which links to the waypoint script 241
[sql.sql] There is no waypoint which links to the waypoint script 242
[sql.sql] There is no waypoint which links to the waypoint script 243
[sql.sql] There is no waypoint which links to the waypoint script 244
[sql.sql] There is no waypoint which links to the waypoint script 245
[sql.sql] There is no waypoint which links to the waypoint script 246
[sql.sql] There is no waypoint which links to the waypoint script 247
[sql.sql] There is no waypoint which links to the waypoint script 248
[sql.sql] There is no waypoint which links to the waypoint script 249
````

**Comment by Killyana** [comment]
Updated

**Comment by mario-64** [comment]
This fix appears to be working fine to me. Can anyone else confirm so it can be merged?

**Comment by tkrokli** [comment]
Using TrinityCore rev. 3aa177410010 2015-06-04 17:01:17 +0200 (3.3.5 branch) (Win64, Release)
DB: `TDB 335.58` with updates up to and including `2015_06_04_01_world.sql`

The 2 NPCs are still only walking, either on top of each other, right next to each other or with Donna ahead of William. Donna does not say much (if anything) and only William appears to talk. No special emotes observed either, so the issue is confirmed on existing source.
***
Running the 3.3.5 SQL script in SQLyog, I get the following output (some blank lines removed) :
```sql
15 queries executed, 15 success, 0 errors, 0 warnings

Query: DELETE FROM `creature_formations` WHERE `leaderGUID`=79720
2 row(s) affected

Execution Time : 0 sec
Transfer Time  : 0.001 sec
Total Time     : 0.002 sec
--------------------------------------------------
Query: INSERT INTO `creature_formations` (`leaderGUID`,`memberGUID`,`dist`,`angle`,`groupAI`) VALUES (79720,79720,4,0,2), (79720,79721,...
2 row(s) affected

Execution Time : 0 sec
Transfer Time  : 0 sec
Total Time     : 0 sec
--------------------------------------------------
Query: DELETE FROM `waypoint_data` WHERE `id`=797210
82 row(s) affected

Execution Time : 0.001 sec
Transfer Time  : 0 sec
Total Time     : 0.001 sec
--------------------------------------------------
Query: DELETE FROM `waypoint_scripts` WHERE `id` BETWEEN 229 AND 270
42 row(s) affected

Execution Time : 0 sec
Transfer Time  : 0 sec
Total Time     : 0 sec
--------------------------------------------------
Query: UPDATE `waypoint_data` SET `action`=0 WHERE `id`=797200
21 row(s) affected

Execution Time : 0 sec
Transfer Time  : 0 sec
Total Time     : 0 sec
--------------------------------------------------
Query: UPDATE `creature_template` SET `MovementType` = 0 WHERE `entry` = 2532
0 row(s) affected

Execution Time : 0 sec
Transfer Time  : 0 sec
Total Time     : 0 sec
--------------------------------------------------
Query: UPDATE `creature_addon` SET `path_id` = 0 WHERE `guid` = 79721
1 row(s) affected

Execution Time : 0 sec
Transfer Time  : 0 sec
Total Time     : 0 sec
--------------------------------------------------
Query: UPDATE `waypoint_data` SET `move_type`=1 WHERE `id`=797200
82 row(s) affected

Execution Time : 0 sec
Transfer Time  : 0 sec
Total Time     : 0.001 sec
--------------------------------------------------
Query: UPDATE `creature` SET `movementtype`=0 WHERE `guid`=79721
1 row(s) affected

Execution Time : 0 sec
Transfer Time  : 0 sec
Total Time     : 0 sec
--------------------------------------------------
Query: UPDATE `creature_template` SET `AIName`='SmartAI' WHERE `entry`=2533
1 row(s) affected

Execution Time : 0 sec
Transfer Time  : 0 sec
Total Time     : 0 sec
--------------------------------------------------
Query: DELETE FROM `smart_scripts` WHERE `entryorguid`=2533 AND `source_type`=0
0 row(s) affected

Execution Time : 0 sec
Transfer Time  : 0 sec
Total Time     : 0 sec
--------------------------------------------------
Query: DELETE FROM `smart_scripts` WHERE `entryorguid`=253300 AND `source_type`=9
0 row(s) affected

Execution Time : 0 sec
Transfer Time  : 0 sec
Total Time     : 0 sec
--------------------------------------------------
Query: INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags...
3 row(s) affected

Execution Time : 0 sec
Transfer Time  : 0 sec
Total Time     : 0 sec
--------------------------------------------------
Query: DELETE FROM `creature_text` WHERE `entry` IN (2533, 2532)
0 row(s) affected

Execution Time : 0 sec
Transfer Time  : 0 sec
Total Time     : 0 sec
--------------------------------------------------
Query: INSERT INTO `creature_text` (`entry`,`groupid`,`id`,`text`,`type`,`language`,`probability`,`emote`,`duration`,`sound`,`comment`,...
16 row(s) affected

Execution Time : 0 sec
Transfer Time  : 0 sec
Total Time     : 0 sec
```
:+1: The script seems completely fine to me so far. Testing in-game, Donna is now talking as well, the pathing seems to work fine, although the run speed is about 1.2 (120%) of the normal run speed of a player character. I suppose it is correct, I just didn't think about it until I had to do `.modify speed all 1.2` to be able to keep up to follow the dialogue. I guess their run speed is intended to be this fast, because children always seem to run around pretty quickly. I just didn't spend much time thinking about the logic. :smirk_cat:

**Comment by DrThum** [comment]
TL;DR for you guys: It's working as intended.

**Comment by tkrokli** [comment]
> TL;DR for you guys: It's working as intended.

Yes, indeed. I should have kept it short, but then you would never know how much time I took testing it.
Also, I should probably have started the post with a warning: " Wall of text crits you for 9000 "
